pilot/pkg/model/push_status.go

```diff
@@ -196,3 +233,154 @@ func (ps *PushStatus) UpdateMetrics() {
 		}
 	}
 }
+
+// VirtualServices lists all virtual services bound to the specified gateways
+// This replaces store.VirtualServices
+func (ps *PushStatus) VirtualServices(gateways map[string]bool) []Config {
+	configs := ps.VirtualServiceConfigs
```

Comment:
env.VirtualServices() could use the global push context - but that's not what we want ( it may change while you are building the config ).

Also - I think config generation should stop using env and any global config - and get everything from the shapshot in push context data structs.